src/main/scala/com/microsoft/hyperspace/index/PlanSignatureProvider.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index
+
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+import com.microsoft.hyperspace.util.HashingUtils
+
+/**
+ * [[PlanSignatureProvider]] provides signature for a logical plan based on
+ * the type of operators in it.
+ * A plan needs to have at least one operator so its signature can be generated.
+ */
+class PlanSignatureProvider extends LogicalPlanSignatureProvider {
+
+  /**
+   * Generate the signature of logical plan.
+   *
+   * @param logicalPlan logical plan.
+   * @return signature if there is at least one operator in the plan; Otherwise None.
+   */
+  def signature(logicalPlan: LogicalPlan): Option[String] = {
+    var signature = ""
+    logicalPlan.foreachUp(o => signature += HashingUtils.md5Hex(o.nodeName))
```

Comment:
could you please add a final HashingUtils.md5Hex(signature) on the overall signature?

Reply:
also `p` instead of `o` as in "plan"?

Reply:
@imback82 `o` is actually added for "Operator" because for each operator we add hash of its node name to the plan.

Reply:
I didn't think "operator" when I saw `o` since technically it's `logicalPlan.foreachUp(p: LogicalPlan => ...)`. But it's nit anyway.

Reply:
Got it, I will change it in the next push.